cowbird/handlers/impl/magpie.py

```diff
@@ -72,34 +98,120 @@ def _send_request(self, method, url, params=None, json=None):
         return resp
 
     def get_service_types(self):
-        # type: () -> List
+        # type: () -> List[str]
         """
         Returns the list of service types available on Magpie.
         """
         if not self.service_types:
             resp = self._send_request(method="GET", url=f"{self.url}/services/types")
             if resp.status_code != 200:
-                raise RuntimeError("Could not get Magpie's services.")
+                raise RuntimeError("Could not get Magpie's service types.")
             self.service_types = list(resp.json()["service_types"])
         return self.service_types
 
     def get_resource_id(self, resource_full_name):
         # type: (str) -> str
         raise NotImplementedError
 
-    def get_resources_tree(self, resource_id):
-        # type: (int) -> List
+    def get_services_by_type(self, service_type):
+        # type: (str) -> Dict[str, JSON]
+        resp = self._send_request(method="GET", url=f"{self.url}/services/types/{service_type}")
+        if resp.status_code != 200:
+            raise RuntimeError(f"Failed to get the services of type `{service_type}`.")
+        return resp.json()["services"][service_type]
+
+    def get_service_info(self, service_name):
+        # type: (str) -> Dict[str, JSON]
+        resp = self._send_request(method="GET", url=f"{self.url}/services/{service_name}")
+        if resp.status_code != 200:
+            raise RuntimeError(f"Could not find the `{service_name}` service info.")
+        return resp.json()["service"]
+
+    def get_resources_by_service(self, service_name):
+        # type: (str) -> Dict[str, JSON]
+        resp = self._send_request(method="GET", url=f"{self.url}/services/{service_name}/resources")
+        if resp.status_code != 200:
+            raise RuntimeError(f"Could not find the `{service_name}` service's resources.")
```

Comment:
There seems to be a mixed use of `RuntimeError` and `HTTPError` for similar checks following an invalid response's status code. This should probably be unified. 

Also, a more specific/derived exception for this handler would be more relevant to indicate a "purposely raised from precheck" error reported in the traceback, rather than a generic runtime issue.

Using this derived exception, it would be valid to add it to the `setup.cfg` in section `[coverage:report]` to ignore those uncovered test cases.